Motion/app/src/main/java/com/example/android/motion/demo/oscillation/CheeseAdapter.kt

```diff
@@ -0,0 +1,192 @@
+
+                override fun onPull(deltaDistance: Float) {
+                    super.onPull(deltaDistance)
+                    handlePull(deltaDistance)
+                }
+
+                override fun onPull(deltaDistance: Float, displacement: Float) {
+                    super.onPull(deltaDistance, displacement)
+                    handlePull(deltaDistance)
+                }
+
+                private fun handlePull(deltaDistance: Float) {
+                    // This is called on every touch event while the list is scrolled with a finger.
+                    // We simply update the view properties without animation.
+                    val sign = if (direction == DIRECTION_RIGHT) -1 else 1
+                    val rotationDelta = sign * deltaDistance * OVERSCROLL_ROTATION_MAGNITUDE
+                    val translationXDelta =
+                        sign * recyclerView.width * deltaDistance * OVERSCROLL_TRANSLATION_MAGNITUDE
+                    recyclerView.forEachVisibleHolder { holder: CheeseViewHolder ->
+                        holder.rotation.cancel()
+                        holder.translationX.cancel()
+                        holder.itemView.rotation += rotationDelta
+                        holder.itemView.translationX += translationXDelta
+                    }
+                }
+
+                override fun onRelease() {
+                    super.onRelease()
+                    // The finger is lifted. This is when we should start the animations to bring
+                    // the view property values back to their resting states.
+                    recyclerView.forEachVisibleHolder { holder: CheeseViewHolder ->
+                        holder.rotation.start()
+                        holder.translationX.start()
+                    }
+                }
+
+                override fun onAbsorb(velocity: Int) {
+                    super.onAbsorb(velocity)
+                    val sign = if (direction == DIRECTION_RIGHT) -1 else 1
+                    // The list has reached the edge on fling.
+                    val translationVelocity = sign * velocity * FLING_TRANSLATION_MAGNITUDE
+                    recyclerView.forEachVisibleHolder { holder: CheeseViewHolder ->
+                        holder.translationX
+                            .setStartVelocity(translationVelocity)
+                            .start()
+                    }
+                }
+            }
+        }
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CheeseViewHolder {
+        return CheeseViewHolder(parent)
+    }
+
+    override fun onBindViewHolder(holder: CheeseViewHolder, position: Int) {
+        val cheese = getItem(position)
+        Glide.with(holder.image).load(cheese.image).into(holder.image)
+        holder.name.text = cheese.name
+        // The rotation pivot should be at the center of the top edge.
+        holder.itemView.doOnNextLayout { v -> v.pivotX = v.width / 2f }
```

Comment:
doOnLayout() should be fine?

Reply:
Yes, replaced this with doOnLayout and moved the logic to onCreateViewHOlder.